### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by carloshdelreal**

### CODE REVIEW

**Review by JamesOkunlade — CHANGES_REQUESTED** [review]
Hello @carloshdelreal , @trox115 ,

There are some major improvements you should make to your project.

- You game only checks for a win and does not check for draw. Usually when the 9 slots get filled and there is no winner, the game should halt with a notice `'it's a draw` but that is not the case in your game. In your case, your game stays on and keep saying a slot is already chosen when one tries to continue playing. Please work on improving this.

- You should ensure the game halts after a winner emerges. You game restart by itself and this is not cool. You should ask the players if they want to play again or quite or better still just make the game stop once there is a winner.

Work on these and request for a re-review

Happy coding!

**Review by JamesOkunlade — CHANGES_REQUESTED** [review]
### Really Great Improvement!!! :grin:

One last little bug you should take few minutes to fix; after winning a game and I get asked if I want to play again, I observed I still get asked if I want to keep players name even after replying with a `N` that I don't want to play again. The game should halt immediately I say I don't want to play again, but apparently, in this case, I still get asked the next question. 

You have done well to improve on my last correction please take some minutes to check this bug as well and request for another review, possibly the last, when you're done.

Happy coding!

File: bin/main.rb [hunk 1]
```diff
@@ -0,0 +1,137 @@
+require '../lib/library.rb'
+class Game
```
Comment (mcrd25) [inline bin/main.rb]: Please try and have this Game class in on file and move it to lib, may have to reconsider some methods and how you implement them. If you need tips, please message me ASAP! I am glad to help!

**Review by mcrd25 — CHANGES_REQUESTED** [review]
# Whoa! great Gameplay! 🥇 

Hi @carloshdelreal and @trox115 👋,

Games works great! However, I'd like to mention a couple of requirements previous reviews did not cover.

### Crucial things to make project works Checklist

- [x] The project has a descriptive README.md
- [x] code is clean: indentation with two spaces, line lengths, spaces around operators
- [x] Ensure that students work with Fit Flow: production `master` branch + main `develop` branch + Pull Requests from feature branches to the `develop`
I noticed you did not use feature branches which you should implement moving forward unto Rails. See [this link](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59882084/)
Now you do not need to address this now but please follow in upcoming projects 😄 
- [x] User interaction commands should be in one executable file main.rb
- [x] Game logic should be kept in classes (no “puts” and “get.chomp” there!) 
You simply added `Game` class to main.rb this does not mean it meets the requirement. Perhaps some refactoring is needed
- [x] The project uses at least two classes (Usually it uses three: Game, Board and Player. The code of the project should be logically divided between classes.
- [x] DRY rule should be respected. E.g. students tend to check the winning position for X and O separately instead of creating a method and reuse it.
- [x] The code should respect the single responsibility rule for methods (i.e. each method does exactly one thing). Students tend to create long methods that are doing multiple steps in the game process.
- [x] Use attr_readers/attr_writers instead of atrr_acessors. The student must be aware that by using it they allow writing that property from outside the class. That variable can be set to anything. 
**Please address this!**
- [x] Don't use hardcoded values for players symbols ("X", "O") to check if a row/column/diagonal has the same elements

**I am really sorry to be a killjoy but I only want your project to meet the necessary standards of a skilled/competent Ruby Developer! I care and would like your project to meet all Requirements laid out by Mciroverse.**

***You really did do a great job though! 👍***

**PLEASE ADDRESS REQUIREMENTS AND SUBMIT ANOTHER REVIEW**

***If you have any questions, let me know here or on slack and I will be glad to clarify with you. You can also reach me and connect with me on my social media.***

**[--Maya](https://github.com/mcrd25)**
[GitHub](https://github.com/mcrd25) [Twitter](https://twitter.com/mcrd25) [linkedin](https://www.linkedin.com/in/mayadouglas/)

**Review by JamesOkunlade — APPROVED** [review]
### Great indeed. It's been a long way down here!!

It would be fair to pass this project now after a couple of refactorings from you guys. However, just before you merge to master, split the `Game` and `Board` classes you have in your library.rb file into each file on its own. i.e. You'll have game.rb and board.rb files instead of the library.rb you have.

effect this and go ahead to merge to master. 

I wish you good luck in your future projects. :+1::+1:

**Pull request merged**